client/www/pages/docs/storage.md

```diff
@@ -5,131 +5,555 @@ title: Storage
 
-{% /callout %}
+```shell {% showCopy=true %}
+npx create-next-app instant-storage --tailwind --yes
+cd instant-storage
+npm i @instantdb/react
+```
+
+Initialize your schema and permissions via the [cli tool](/docs/cli)
 
-## Uploading files
+```
+npx instant-cli@latest init
+```
 
-We use the `db.storage.upload(pathname: string, file: File)` function to upload a file.
+Now open `instant.perms.ts` and add the following permissions
 
-```tsx
-async function upload(files: FileList) {
-  const file = files[0];
-  // use the file's current name as the path
-  await db.storage.upload(file.name, file);
-  // or, give the file a custom name
-  await db.storage.upload('demo.png', file);
-  // or, put it in the `images` subdirectory
-  await db.storage.upload('images/demo.png', file);
-  // or, put it in a subdirectory for the current user,
-  // and restrict access to this file via Storage permissions
-  await db.storage.upload(`${currentUser.id}/demo.png`, file);
+```javascript {% showCopy=true %}
+// Note:
+// For production apps you should use more restrictive permissions
+// But for this quick start example we'll allow anyone to view, upload,
+// and delete files
+{
+  "$files": {
+    "allow": {
+      "view": "true",
+      "create": "true",
+      "delete": "true"
+    }
+  }
+}
+```
+
+And then replace the contents of `app/src/page.tsx` with the following code
+
+```javascript {% showCopy=true %}
+'use client';
+
+import { init } from '@instantdb/react';
+import schema from '../instant.schema';
+import React from 'react';
+
+// Types
+// ----------
+export type Image = {
+  id: string;
+  path: string;
+  url: string;
+};
+
+// Instant app
+const APP_ID = '__APP_ID__';
+
+const db = init({ appId: APP_ID, schema });
+
+// `uploadFile` is what we use to do the actual upload!
+// the `$files` will automatically update once the upload is complete
+async function uploadImage(file: File) {
+  try {
+    // Optional metadata you can set for uploads
+    const opts = {
+      // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type
+      // Default: 'application/octet-stream'
+      contentType: file.type,
+      // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition
+      // Default: 'inline'
+      contentDisposition: 'attachment; filename="moop.jpg"',
+    };
+    await db.storage.uploadFile(file.name, file, opts);
+  } catch (error) {
+    console.error('Error uploading image:', error);
+  }
+}
+
+// `delete` is what we use to delete a file from storage
+// `$files` will automatically update once the delete is complete
+async function deleteImage(image: Image) {
+  await db.storage.delete(image.path);
+}
+
+function App() {
+  // $files is the special namespace for querying storage data
+  const { isLoading, error, data } = db.useQuery({
+    $files: {
+      $: {
+        order: { serverCreatedAt: 'asc' },
+      },
+    },
+  });
+
+  if (isLoading) {
+    return <div>Fetching data...</div>;
```

Comment:
nit: I would return `null` for isLoading. 

Reasoning: Otherwise this will have a bit of a jarring flash when you first load the page